analyzers/tests/SonarAnalyzer.Test/TestCases/SymbolicExecution/Roslyn/ConditionEvaluatesToConstant.cs

```diff
@@ -4321,10 +4321,67 @@ public void IndexerReturnsNullInsteadOfThrowingException(NameValueCollection col
         if (element != null)
         {
             Console.WriteLine(element.ToString());
+            if (collection.Count == 0)          // Noncompliant - the indexer returned a non-null result, so the collection is not empty
+            {
+                Console.WriteLine("Empty!");    // Secondary
+            }
         }
-        if (collection.Count == 0)          // Noncompliant - FP: the indexer with string argument returns null if the key is not found rather than throwing an exception,
+        if (collection.Count == 0)          // Noncompliant - FP: the indexer returns null if the key is not found rather than throwing an exception,
         {                                   // so at this point we can't know for sure that the collection is not empty.
             Console.WriteLine("Empty!");    // Secondary - FP
         }
     }
+
+    public void IndexerReturnsNullInsteadOfThrowingException(PersonCollection collection)
```

Comment:
Very good example. It could be much simpler if `Refresh()` would just clear out the collection no matter what. On the other hand, this is more of a real-world example, which is also nice.